include/deal.II/fe/fe.h

```diff
@@ -654,9 +654,17 @@ class FiniteElement : public Subscriptor,
    * ExcShapeFunctionNotPrimitive. In that case, use the
    * shape_value_component() function.
    *
-   * An ExcUnitShapeValuesDoNotExist is thrown if the shape values of the
-   * FiniteElement under consideration depends on the shape of the cell in
-   * real space.
+   * Implementations of this function should throw an exception of type
+   * ExcUnitShapeValuesDoNotExist if the shape functions of the
+   * FiniteElement under consideration depend on the shape of the cell in
+   * real space, i.e., if the shape functions are not defined by
+   * mapping from the reference cell. Some non-conforming elements are
+   * defined this way, as is the FE_DGPNonparametric class, to name just
+   * one example.
```

Comment:
I looked through the code for `FE_DGPNonparametric` and it looks like this function will not raise this exception (it ultimately resolves to a call to ``PolynomialSpace<dim>::compute_value` if I have followed things correctly). I think this means we need a different example.